### PR TITLE
E2E test with JetStream enabled Nats cluster

### PR DESCRIPTION
**Pull request by boecklim**
- NATS JetStream cluster in E2E tests

### CODE REVIEW

File: test/config/config.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ prometheusEndpoint: /metrics
 prometheusAddr: :2112
 grpcMessageSize: 100000000
 network: regtest
-queueURL: nats://nats:4222
+queueURL: nats://nats-1:4222,nats://nats-2:4223,nats://nats-3:4224
```
Comment (kuba-4chain) [inline test/config/config.yaml:9]: The PR is looking great 👍 

I just have one question, why do we need 3 instances of NATS? Is it to make sure no messages are lost in case one NATS instance failing/disconnecting? 

Even if so, it should be extremely rare for that to happen, so maybe it's not worth the resources and time for e2e tests to spin up 3 instances? One should be enough? WDYT?

Comment (boecklim) [reply test/config/config.yaml:9]: That is right. Just one would be enough for sure.
I just thought it is good to have a cluster here in the e2e test because that also allows to easily lookup how a cluster needs to configured for anybody who wants to run ARC with a JetStream-enabled NATS cluster. So It serves a kind of documentation purposes. What do you think?

Comment (arkadiuszos4chain) [reply test/config/config.yaml:9]: I think it was a good idea to use a cluster, but @kuba-4chain has a point. Maybe we should limit instances to 2 for easier and faster running tests on local machines.

Maybe we could also apply the same approach to other services?

File: test/docker-compose.yml [hunk 2]
```diff
@@ -101,15 +101,47 @@ services:
...
 
-  nats:
+  nats-1:
     image: nats:2.10.10
-    container_name: 'nats-server'
-    restart: on-failure
-    expose:
-      - "4222"
+    container_name: nats-server-1
+    restart: no
```
Comment (arkadiuszos4chain) [inline test/docker-compose.yml:107]: Why did you change `restart` option?

File: test/config/config.yaml (hunk 1 above)
Comment (boecklim) [reply test/config/config.yaml:9]: Ok, I'll reduce it to 2 instances.

@arkadiuszos4chain Which other services do you mean?

File: test/docker-compose.yml (hunk 2 above)
Comment (boecklim) [reply test/docker-compose.yml:107]: @arkadiuszos4chain My idea was that the whole e2e test should fail if for any reason of misconfiguration of NATS stops and NATS shouldn't restart in that case. But there is not a strong reason to change it. Could also revert the change.

Comment (arkadiuszos4chain) [reply test/docker-compose.yml:107]: I was just curious. Thanks for the clarification

File: test/config/config.yaml (hunk 1 above)
Comment (arkadiuszos4chain) [reply test/config/config.yaml:9]: Nodes and blocktx, I think we are running 3 nodes for tests and scale blocktx to 7 instances

Comment (boecklim) [reply test/config/config.yaml]: @arkadiuszos4chain Sounds good. But let's do it in a separate PR

**Review by kuba-4chain — APPROVED** [review]
